src/compiler/checker.ts

```diff
@@ -19518,6 +19518,21 @@ namespace ts {
             return type;
         }
 
+        function checkNonNullNonVoidType(
+            type: Type,
+            node: Node,
+            nullDiagnostic?: DiagnosticMessage,
+            undefinedDiagnostic?: DiagnosticMessage,
+            nullOrUndefinedDiagnostic?: DiagnosticMessage,
+            voidDiagnostic?: DiagnosticMessage
+        ): Type {
+            const nonNullType = checkNonNullType(type, node, nullDiagnostic, undefinedDiagnostic, nullOrUndefinedDiagnostic);
+            if (nonNullType !== errorType && nonNullType.flags & TypeFlags.Void) {
+                error(node, voidDiagnostic || Diagnostics.Object_is_possibly_undefined);

```

Comment:
Not sure if this is the right diagnostic message?

Reply:
Possibly-undefined is the right error state for this condition. `void` means "not specified" and is thus correctly "possibly undefined"